database/errors_test.go

```diff
@@ -264,3 +264,79 @@ func TestDatabaseToApiError(t *testing.T) {
 		})
 	}
 }
+
+type MockAssertion struct{}
+
+func (ma *MockAssertion) EqualError(theError error, errString string, msgAndArgs ...interface{}) bool {
+	if nil == theError {
+		return false
+	}
+	return theError.Error() == errString
+}
+
+func TestEqualLogError(t *testing.T) {
+	tests := []struct {
+		name     string
+		err      error
+		expected string
+		equal    bool
+	}{
+		{
+			name:     "nil error",
+			err:      nil,
+			expected: "",
+			equal:    false,
+		},
+		{
+			name:     "line number match",
+			err:      errors.New("[GAD.DAT.123]"),
+			expected: "[GAD.DAT.123]",
+			equal:    true,
+		},
+		{
+			name:     "line number mismatch",
+			err:      errors.New("[GAD.DAT.123]"),
+			expected: "[GAD.DAT.456]",
+			equal:    true,
+		},
+		{
+			name:     "file prefix mismatch",
+			err:      errors.New("[GAD.DAT.123]"),
+			expected: "[GAD.QB.123]",
+			equal:    false,
+		},
+		{
+			name:     "short prefix",
+			err:      errors.New("foo[A.B.1]bar"),
+			expected: "foo[A.B.2]bar",
+			equal:    true,
+		},
+		{
+			name:     "error message",
+			err:      errors.New("foo[GAD.DAT.123]bar"),
+			expected: "foo[GAD.DAT.567]bar",
+			equal:    true,
+		},
+		{
+			name:     "error message mismatch",
+			err:      errors.New("foo[GAD.DAT.123]bar"),
+			expected: "foo[GAD.DAT.567]baz",
+			equal:    false,
+		},
```

Comment:
Missing tests for conditions: 
1. No brackets in the string e.g. "foo bar"
2. Multiple brackets e.g. "foo bar [ABC.DEF.123] [QWE.ZXC.256]"

It appears that they should pass when you add them.

Reply:
Also missing empty string tests, e.g. errors.New("")

Reply:
Updated, thanks!